src/main/groovy/io/seqera/wave/controller/MetricsController.groovy

```diff
@@ -0,0 +1,168 @@
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package io.seqera.wave.controller
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalTime
+import java.time.ZoneId
+import java.time.format.DateTimeParseException
+import javax.annotation.Nullable
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.MediaType
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Error
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.PathVariable
+import io.micronaut.http.annotation.QueryValue
+import io.micronaut.security.annotation.Secured
+import io.micronaut.security.authentication.AuthorizationException
+import io.micronaut.security.rules.SecurityRule
+import io.seqera.wave.service.metric.Metric
+import io.seqera.wave.service.metric.MetricFilter
+import io.seqera.wave.service.metric.MetricService
+import jakarta.inject.Inject
+import static io.micronaut.http.HttpHeaders.WWW_AUTHENTICATE
+
+/**
+ * Controller for wave Metrics
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+@Slf4j
+@CompileStatic
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/v1alpha1/metrics")
+class MetricsController {
+    @Inject
+    private MetricService metricsService
+
+    @Get(uri = "/builds/{metric}", produces = MediaType.APPLICATION_JSON)
+    HttpResponse<Map> getBuildMetrics(@PathVariable String metric,
+                                      @Nullable @QueryValue Boolean success,
+                                      @Nullable @QueryValue String startDate,
+                                      @Nullable @QueryValue String endDate,
+                                      @Nullable @QueryValue Integer limit) {
+        return HttpResponse.ok(
+                metricsService.getBuildMetrics(
+                        Metric.valueOf(metric),
+                        new MetricFilter.Builder()
+                                .dates(parseStartDate(startDate), parseEndDate(endDate))
+                                .success(success)
+                                .limit(limit)
+                                .build()
+                )
+        )
+    }
+
+    @Get(uri = "/builds", produces = MediaType.APPLICATION_JSON)
+    HttpResponse<LinkedHashMap> getBuildsCount(@Nullable @QueryValue Boolean success,
+                                               @Nullable @QueryValue String startDate,
+                                               @Nullable @QueryValue String endDate) {
+        return HttpResponse.ok(
+                [count: metricsService.getBuildCount(
+                        new MetricFilter.Builder()
+                                .dates(parseStartDate(startDate), parseEndDate(endDate))
+                                .success(success)
+                                .build())
+                ]
+        )
```

Comment:
Let's use a proper `GetBuildsCountsResponse` object. 

Same for all the remaining endpoints

Reply:
done